modules/user/lib/user-balance.service.ts

```diff
@@ -10,30 +10,49 @@ export class UserBalanceService {
     constructor() {}
 
     public async getUserPoolBalances(address: string, chains: Chain[]): Promise<UserPoolBalance[]> {
-        const user = await prisma.prismaUser.findUnique({
-            where: { address: address.toLowerCase() },
-            include: {
-                walletBalances: {
-                    where: { chain: { in: chains }, poolId: { not: null }, balanceNum: { gt: 0 } },
-                },
-                stakedBalances: {
-                    where: { chain: { in: chains }, poolId: { not: null }, balanceNum: { gt: 0 } },
-                },
+        // this seems to cause heavy load
+        // const user = await prisma.prismaUser.findUnique({
+        //     where: { address: address.toLowerCase() },
+        //     include: {
+        //         walletBalances: {
+        //             where: { chain: { in: chains }, poolId: { not: null }, balanceNum: { gt: 0 } },
+        //         },
+        //         stakedBalances: {
+        //             where: { chain: { in: chains }, poolId: { not: null }, balanceNum: { gt: 0 } },
+        //         },
+        //     },
+        // });
+
+        const userWalletBalances = await prisma.prismaUserWalletBalance.findMany({
+            where: {
+                userAddress: address.toLowerCase(),
+                chain: { in: chains },
+                poolId: { not: null },
+                balanceNum: { gt: 0 },
```

Comment:
Do we need to add these two conditions to the query? I wonder when pool can be empty. And empty balances can be filtered out in JS. Otherwise it needs an index to make searching faster and adding it to two columns will make writing slower (probably we can live with it, but better to avoid if not needed)

Reply:
we want to add the balance > 0, as otherwise all pools would be returned that the user was ever part of.

true about the pool Id, can probably remove.

Reply:
let's remove balance as well to keep the queries simple, frontend can filter 0 balances out if needed

Reply:
Filtering non-zero balances manually now.